src/Missings.jl

```diff
@@ -208,6 +208,448 @@ missing
 """
 passmissing(f) = PassMissing{Core.Typeof(f)}(f)
 
+abstract type AbstractSpread end
+struct SpreadDefault <: AbstractSpread end
+struct SpreadNonMissing <: AbstractSpread end
+struct SpreadNone <: AbstractSpread end
+struct SpreadAll <: AbstractSpread end
+
+struct SpreadMissings{F, S <: AbstractSpread} <: Function
+    f::F
+    spread::S
+    function SpreadMissings(f, spread::AbstractSpread)
+        if !(spread isa AbstractSpread)
+            throw(ArgumentError("spread must be either SpreadDefault(), SpreadNonMissing(), or SpreadNone()"))
+        end
+        new{Core.Typeof(f), typeof(spread)}(f, spread)
+    end
+end
+
+"""
+    nomissing_subarray(a::AbstractVector, nonmissinginds::AbstractVector)
+
+Given an input vector `a` where `nonmissinginds` is guaranteed
+to not include any missing values, return a `SubArray` referencing
+the `nonmissinginds`. The element type of the returned output
+does not include `missing`.
+"""
+function nomissing_subarray(a::AbstractVector, nonmissinginds::AbstractVector)
+    T = nonmissingtype(eltype(a)) # Element type
+    N = 1 # Dimension of view
+    P = typeof(a) # Type of parent array
+    I = Tuple{typeof(nonmissinginds)} # Type of the non-missing indices
+    L = Base.IndexStyle(a) === IndexLinear # If the type supports fast linear indexing
+    SubArray{T, N, P, I, L}(a, (nonmissinginds,), 0, 1)
```

Comment:
The docstring for `SubArray` makes no mention of contructors, and in fact says "Construct `SubArray`s using the `view` function." So this direct usage of a `SubArray` constructor is usage of undocumented Base internals that could change in any minor or patch release.

Reply:
Yeah, that's probably true. 

I think it will be worthwhile to make our own array type that does this better... hopefully it's not too hard.